### ISSUE TITLE
Configure/start service after main() => runApp

### ISSUE THREAD

**Issue by carman247** [ekasetiawans/flutter_background_service#186]
Is it possible to start the background service after the `main` `runApp()` function?

e.g. 

Could the service be started in `initState` for the `home` widget of `MaterialApp`

**Comment by ekasetiawans** [comment]
@carman247 it's possible, but I recommend running the service on the main method immediately. Regardless you will execute something on the service or not. You can start running a task on initState using the invoke method.

**Comment by carman247** [comment]
@ekasetiawans ok thanks for the quick response. I think I realised I need to run in the main anyway as seems like the config require "top level" functions, and I was hoping to use functions in a `Provider`.

**Comment by ekasetiawans** [comment]
@carman247 you cant't do that because background service running in different isolate, so you will never able to get the provider instance.

**Comment by vauvenal5** [comment]
Hi!

I run in a similar issue. I was  starting and stopping the service when the user was downloading files. However, I noticed that when starting the service after runApp and debugger was attached it restarted the main UI thread without any error message or anything.

Currently I switched to starting the service before runApp and just moving it to the foreground when the user is downloading something and the back into the background. Is this actually the way how it is expected to use it?

Thx!

**Comment by vauvenal5** [comment]
After some more investigation, the issue also happens when starting the background service before runApp but then it does not hurt. However, I do not know what triggers it.

TL;DR When a debugger attached starting the background service leads to a reload of the main thread. Does not happen without attached debugger.
